contracts/deposit-contract-tests/src/test_fixture.rs

```diff
@@ -0,0 +1,244 @@
+use casper_types::{
+    account::AccountHash, contracts::NamedKeys, runtime_args, ContractHash, Key, RuntimeArgs, URef,
+    U512,
+};
+use lazy_static::lazy_static;
+use utils::create_funded_dummy_account;
+extern crate dotenvy;
+use dotenvy::dotenv;
+use std::env;
+
+pub const ACCOUNT_USER_1: [u8; 32] = [1u8; 32];
+pub const ACCOUNT_USER_2: [u8; 32] = [2u8; 32];
+pub const ACCOUNT_USER_3: [u8; 32] = [3u8; 32];
+
+// This defines a static variable for the path to WASM binaries
+lazy_static! {
+    static ref PATH_TO_WASM_BINARIES: String = {
+        dotenv().ok(); // Load the .env file at runtime
+        env::var("PATH_TO_WASM_BINARIES").expect("Missing environment variable PATH_TO_WASM_BINARIES")
+    };
+}
+
+#[cfg(test)]
+pub struct TestContext {
+    pub builder: InMemoryWasmTestBuilder,
+    pub account_1: AccountHash,
+    pub account_2: AccountHash,
+    pub account_3: AccountHash,
+}
+
+impl TestContext {
+    pub fn new() -> TestContext {
+        let mut builder = InMemoryWasmTestBuilder::default();
+        builder.run_genesis(&PRODUCTION_RUN_GENESIS_REQUEST);
+        let account_1 = create_funded_dummy_account(&mut builder, Some(ACCOUNT_USER_1));
+        let account_2 = create_funded_dummy_account(&mut builder, Some(ACCOUNT_USER_2));
+        let account_3 = create_funded_dummy_account(&mut builder, Some(ACCOUNT_USER_3));
+
+        TestContext {
+            builder,
+            account_1,
+            account_2,
+            account_3,
+        }
+    }
+
+    pub fn named_keys(&self, account: AccountHash) -> NamedKeys {
+        self.builder
+            .get_expected_account(account)
+            .named_keys()
+            .clone()
+    }
+
+    pub fn contract_named_keys(
+        &self,
+        contract_name: &str,
+        key_name: &str,
+        account: AccountHash,
+    ) -> Key {
+        let contract_hash = self.contract_hash_from_named_keys(contract_name, account);
+        *self
+            .builder
+            .get_contract(contract_hash)
+            .expect("should have contract")
+            .named_keys()
+            .get(key_name)
+            .unwrap()
+    }
+
+    pub fn contract_hash_from_named_keys(
+        &self,
+        key_name: &str,
+        account: AccountHash,
+    ) -> ContractHash {
+        self.named_keys(account)
+            .get(key_name)
+            .expect("must have contract hash key as part of contract creation")
+            .into_hash()
+            .map(ContractHash::new)
+            .expect("must get contract hash")
+    }
+
+    pub fn contract_hash(&self, name: &str, account: AccountHash) -> ContractHash {
+        self.builder
+            .get_expected_account(account)
+            .named_keys()
+            .get(name)
+            .expect("must have contract hash key as part of contract creation")
+            .into_hash()
+            .map(ContractHash::new)
+            .expect("must get contract hash")
+    }
+
+    pub fn install(&mut self, admin: AccountHash) {
```

Comment:
Why we need admins for the contract? I believe there is no need for ACL, and it could be simplified a lot.

Reply:
The admin will be able to withdraw from the deposit contract purse. I like the SecurityBadges.

Reply:
This is big security risk, as it makes Kairos centralized solution. There MIGHT be a specific reason to include admins in L1 contract, but "liking" it is not enough to justify it.

Reply:
But this is exactly the design we discussed in our calls. We are planning to build a centralized demo with one single service/node. I don't think we changed the design spec since but if we did then nobody told me about it.

Reply:
I removed the withdrawal functionality since it is not part of the spec currently, but the admin updates the sliding window pointer. This was discussed in our flow chart sessions with Marijan and Avi.

Reply:
We could decide to get rid of the pointer / it could be stored on the centralized L2 server, but that would have to be discussed with the others since it deviates from our design.

We are not building a decentralized solution because we would need a decentralized sequencer for that. V0.1 of Kairos will be centralized and we have discussed this many times.